File: leetcode/go/136.single-number.go
/*
 * @lc app=leetcode id=136 lang=golang
 *
 * [136] Single Number
 */

// @lc code=start
func singleNumber(nums []int) int {
	length := len(nums)
	ret := 0
	for i := 0; i < length; i++ {
		ret = ret ^ nums[i]
	}
	return ret
}

// @lc code=end

File: leetcode/cpp/136.single-number.cpp
/*
 * @lc app=leetcode id=136 lang=cpp
 *
 * [136] Single Number
 */

#include<iostream>
#include <vector>
using namespace std;
// @lc code=start
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        int ret = nums[0];
       for(int i = 1; i < nums.size(); ++i){
        ret = ret ^ nums[i];
       }
       return ret;
    }
};
// @lc code=end

